api/authenticator.py

```diff
@@ -1092,12 +1092,12 @@ def create_authentication_document(self):
             doc['color_scheme'] = description
 
         # Add the library's web colors, if it has any.
-        background = ConfigurationSetting.for_library(
-            Configuration.WEB_BACKGROUND_COLOR, library).value
-        foreground = ConfigurationSetting.for_library(
-            Configuration.WEB_FOREGROUND_COLOR, library).value
-        if background or foreground:
-            doc["web_color_scheme"] = dict(background=background, foreground=foreground)
+        primary = ConfigurationSetting.for_library(
+            Configuration.WEB_PRIMARY_COLOR, library).value
+        secondary = ConfigurationSetting.for_library(
+            Configuration.WEB_SECONDARY_COLOR, library).value
+        if primary or secondary:
+            doc["web_color_scheme"] = dict(primary=primary, secondary=secondary)
```

Comment:
You could restore backwards compatibility by putting the same colors in here under `background` and `foreground`. We can take it out later. Doing this would make it possible to merge the branch without waiting to hear from everyone involved.

Reply:
You'd also need a migration script to move the old colors over rather than resetting everything to the default. I will need to be able to deploy a new Open eBooks circulation manager without first upgrading the web catalog, so even though this is a pain for you I'd like to see it in this branch.

Reply:
The migration script might mean that you start out with a primary and secondary color that don't pass the contrast-with-white test, which I think is better than the alternative. If you disagree, then I'm OK with skipping the migration script. So long as you keep `background` and `foreground` in `web_color_scheme` I can hack the database to get Open eBooks web catalog to continue working.

Reply:
Okay I agree with you, we can add the old values at this point in the CM for backwards compat, but this will only work if you don't edit them, because once you edit them, they will need to pass the contrast with white check. We will also need to let people know that when they deploy a new instance of CPW, they need to update these settings in their CM if they had ones set previously.